### PR TITLE
Mentor application form

### PR DESCRIPTION
**Pull request by apanjain**

### CODE REVIEW

File: backend/mentors/models.py [hunk 1]
```diff
@@ -139,3 +141,48 @@ class MentorPlacement(models.Model):
...
+        max_length=1000,
+        blank=True,
+        null=True
+    )
+    qualities = models.TextField(
+        max_length=1000,
+        blank=True,
+        null=True
+    )
+    mobile = models.IntegerField(
```
Comment (hacksdump) [inline backend/mentors/models.py:181]: Write a phone number validator. We'll assume all phone numbers are exactly 10-digit long.

File: backend/mentors/models.py [hunk 2]
```diff
@@ -139,3 +141,48 @@ class MentorPlacement(models.Model):
...
+        max_length=200,
+        default="",
+        unique=True
+    )
+    name = models.CharField(
+        max_length=1000,
+        blank=True,
+        null=True
+    )
+    enrollno = models.IntegerField(
```
Comment (hacksdump) [inline backend/mentors/models.py:157]: Write an enrolment number validator.

File: backend/mentors/views.py [hunk 3]
```diff
@@ -102,3 +106,23 @@ def create(self, request):
             'interest_ids': interest_ids
         }
         return Response(data, status=status.HTTP_201_CREATED)
+
+
+class MentorApplicationView(APIView):
+
+    def post(self, request):
+        r = requests.post(
```
Comment (hacksdump) [inline backend/mentors/views.py:114]: Separate captcha validation to separate function call.

File: backend/pages/utils.py [hunk 4]
```diff
@@ -1,8 +1,130 @@
...
+from uuid import uuid4
+
+from django.utils.text import slugify
+from django.utils.deconstruct import deconstructible
+
+from backend.settings import MEDIA_ROOT
+
+# Rename file to id of instance
+@deconstructible
+class UploadToPathAndRename(object):
```
Comment (hacksdump) [inline backend/pages/utils.py:12]: Create an abstract/parent class for this functionality and inherit from it to create these kinds of classes.

File: frontend/src/routes/DataCollection/MentorApplicationForm.js [hunk 5]
```diff
@@ -0,0 +1,342 @@
+import React, { Component } from "react";
+import ReCAPTCHA from "react-google-recaptcha";
+import Button from "../../components/Button";
+import styles from "./MentorApplicationForm.module.scss";
+import Select from "react-select";
+import LoadingOverlay from "components/LoadingOverlay";
+import { Redirect } from "react-router-dom";
+import { postMentorApplication } from "api/methods";
+
+const yearOptions = [
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorApplicationForm.js:10]: Move this const to a commonly accessible file and import wherever required.

File: frontend/src/routes/DataCollection/MentorApplicationForm.js [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+              mobile: "",
+              resume: null,
+              redirect: true,
+              isLoading: false,
+            });
+          }
+        })
+        .catch((error) => {
+          const errorMsg =
+            "There was an error while submitting your application.\nPlease try again later";
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorApplicationForm.js:144]: Fetch more descriptive error messages from backend.

File: frontend/src/routes/DataCollection/MentorApplicationForm.js [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+                name="qualities"
+                value={this.state.qualities}
+                id="qualities"
+                onChange={this.handleChange}
+                rows="4"
+                required
+              />
+            </div>
+
+            <div className={styles["form-group"]}>
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorApplicationForm.js:288]: A lot of these input components are similar to mentor detail editor form. I'd highly recommend creating separate components for inputs of each type.

Comment (apanjain) [reply frontend/src/routes/DataCollection/MentorApplicationForm.js:288]: Still we we'll have to pass almost all the same number of props to the input components(name,value, placeholder,onChange etc). Code will still be the same.

**Pull request closed without merging**